packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/helpers/validation.test.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { loggerMock, type MockedLogger } from '@kbn/logging-mocks';
+import type {
+  ISavedObjectTypeRegistry,
+  SavedObjectsModelVersion,
+  SavedObjectsType,
+} from '@kbn/core-saved-objects-server';
+// import {
+//   SavedObjectsTypeValidator,
+//   modelVersionToVirtualVersion,
+// } from '@kbn/core-saved-objects-base-server-internal';
+import { type SavedObjectSanitizedDoc } from '@kbn/core-saved-objects-server';
+import { ValidationHelper } from './validation';
+import { schema } from '@kbn/config-schema';
+
+const defaultVersion = '8.7.0';
+const type = 'my-type';
+
+const createTypeRegistry = () => {
```

Comment:
very much WIP

Reply:
For such unit tests, it's usually easier to work with a real  `SavedObjectTypeRegistry` instance instead of trying to stub/mock `ISavedObjectTypeRegistry`

Reply:
We're not directly testing the registry, hence using a mock. 

However, indirectly, we are, and, since we're missing support for SO models, using the real instance gives us a bit more surety everything works as intended.

I changed to using the real instance for those reasons.